### PR TITLE
'window.close' tracking

### PR DESCRIPTION
**Pull request by miherlosev**

### CODE REVIEW

**Comment by miherlosev** [comment]
@testcafe-build-bot retest

**Comment by miherlosev** [comment]
@testcafe-build-bot retest

File: src/client/driver/driver.js [hunk 1]
```diff
@@ -248,17 +254,52 @@ export default class Driver extends serviceUtils.EventEmitter {
...
         this.childWindowDriverLinks.push(childWindowDriverLink);
+        this._ensureChildWindowClosedWatcher();
+    }
+
+    _ensureChildWindowClosedWatcher () {
+        if (this.checkChildWindowClosedIntervalId)
+            return;
+
+        this.checkChildWindowClosedIntervalId = nativeMethods.setInterval.call(window, () => {
+            for (const childWindowDriverLink of this.childWindowDriverLinks) {
```
Comment (AlexKamaev) [inline src/client/driver/driver.js]: will it work in IE?

File: src/client/driver/driver.js [hunk 2]
```diff
@@ -248,17 +254,52 @@ export default class Driver extends serviceUtils.EventEmitter {
...
+                if (childWindowDriverLink.driverWindow.closed) {
+                    arrayUtils.remove(this.childWindowDriverLinks, childWindowDriverLink);
+
+                    if (this.role === DriverRole.replica)
+                        this._setCurrentWindowAsMaster();
```
Comment (AlexKamaev) [inline src/client/driver/driver.js]: This code supposes that we can have multiple `childWindowDriverLinks` and I suppose that in some cases we call `this._setCurrentWindowAsMaster` multiple times. Or not?

File: src/client/driver/driver.js [hunk 3]
```diff
@@ -248,17 +254,52 @@ export default class Driver extends serviceUtils.EventEmitter {
...
+        this._ensureChildWindowClosedWatcher();
+    }
+
+    _ensureChildWindowClosedWatcher () {
+        if (this.checkChildWindowClosedIntervalId)
+            return;
+
+        this.checkChildWindowClosedIntervalId = nativeMethods.setInterval.call(window, () => {
+            for (const childWindowDriverLink of this.childWindowDriverLinks) {
+                if (childWindowDriverLink.driverWindow.closed) {
```
Comment (AlexKamaev) [inline src/client/driver/driver.js]: we can put this condition in `.filter` function

File: src/client/driver/driver.js (hunk 1 above)
Comment (miherlosev) [reply src/client/driver/driver.js]: We have a special babel transpiler for the `for-of` loop: https://github.com/DevExpress/testcafe/blob/master/package.json#L61

The following code:
```js
for (const childWindowDriverLink of this.childWindowDriverLinks) {
```

transpiled to
```js
for (var _i2 = 0, _childWindowDriverLin2 = _this2.childWindowDriverLinks, _length2 = _childWindowDriverLin2.length; _i2 < _length2; _i2++) {
var childWindowDriverLink = _childWindowDriverLin2[_i2];
```
```

File: src/client/driver/driver.js (hunk 3 above)
Comment (miherlosev) [reply src/client/driver/driver.js]: Agree. I will change this code.

File: src/client/driver/driver.js (hunk 2 above)
Comment (miherlosev) [reply src/client/driver/driver.js]: Yes, you are right. I will fix it.

**Comment by miherlosev** [comment]
@testcafe-build-bot retest

**Comment by miherlosev** [comment]
@testcafe-build-bot retest

**Comment by miherlosev** [comment]
@AlexKamaev @kirovboris FPR

**Pull request merged**